src/main/java/com/minecolonies/blocks/BlockBarrel.java

```diff
@@ -0,0 +1,138 @@
+package com.minecolonies.blocks;
+
+import com.minecolonies.creativetab.ModCreativeTabs;
+import com.minecolonies.items.ModItems;
+import com.minecolonies.lib.Constants;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.init.Items;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.BlockPos;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+import net.minecraft.block.Block;
+import java.util.Random;
+
+/**
+ * Created by Northstar on 8/29/2016.
+ */
+
+public class BlockBarrel extends Block
+{
+    public int timer=0;
+
+    public int BarrelState=0;
+    //barrel's fullnes state
+    private  int fullnes =0;
+    //is compost cooked ?
+    private  boolean isCompostReady =false;
+    /**
+     * The hardness this block has.
+     */
+    private static final float  BLOCK_HARDNESS = 5F;
+
+    /**
+     * This blocks name.
+     */
+    private static final String BLOCK_NAME     = "blockBarrel";
+
+    /**
+     * The resistance this block has.
+     */
+    private static final float  RESISTANCE     = 1F;
+
+
+    public BlockBarrel()
+    {
+        super(Material.wood);
+        this.setTickRandomly(true);
+        initBlock();
+    }
+
+    /**
+     * initialize the block
+     */
+    private void initBlock()
+    {
+        setRegistryName(BLOCK_NAME);
+        setUnlocalizedName(String.format("%s.%s", Constants.MOD_ID.toLowerCase(), BLOCK_NAME));
+        setCreativeTab(ModCreativeTabs.MINECOLONIES);
+        GameRegistry.registerBlock(this, BLOCK_NAME);
+        setHardness(BLOCK_HARDNESS);
+        setResistance(RESISTANCE);
+    }
+
+    /**
+     * Used to determine ambient occlusion and culling when rebuilding chunks for render.
+     *
+     * @return true
+     */
+    @Override
+    public boolean isOpaqueCube()
+    {
+        return true;
+    }
+
+    //whenever player right click to barrel call this.
+    public boolean onBlockActivated(World worldIn, BlockPos pos, IBlockState state, EntityPlayer playerIn, EnumFacing side, float hitX, float hitY, float hitZ)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "onBlockActivated" is 19 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1161)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)